### PR TITLE
Pin `time` to <0.3.12 to ensure we stay no-std compatible

### PR DESCRIPTION
**Pull request by romac**
* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Added entry in `.changelog/`

### CODE REVIEW

**Comment by tarcieri** [comment]
What changed upstream?

**Comment by romac** [comment]
@tarcieri Unclear but we suspect that this is the offending PR: https://github.com/time-rs/time/pull/491

It bumped `js-sys` to 0.3.58` in our dependency tree, and that version apparently breaks the no-std/wasm checks.

From Slack:
> Interestingly, if I explicitly add `js-sys` as a dependency in `tools/no-std-check/Cargo.toml` and restrict its version to `js-sys = { version = ">=0.3.0,<0.3.58" }` it passes. It seems like `js-sys` v0.3.57 compiles just fine, but from v0.3.58 onwards fails.

File: light-client/Cargo.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ serde_cbor = { version = "0.11.1", default-features = false, features = ["alloc"
 serde_derive = { version = "1.0.106", default-features = false }
 sled = { version = "0.34.3", optional = true, default-features = false }
 static_assertions = { version = "1.1.0", default-features = false }
-time = { version = "0.3", default-features = false, features = ["std"] }
+time = { version = ">=0.3, <0.3.12", default-features = false, features = ["std"] }
```
Comment (thanethomson) [inline light-client/Cargo.toml:50]:
```suggestion
# TODO(thane): Remove restrictions once js-sys issue is resolved (causes the Substrate no_std check to fail)
time = { version = ">=0.3, <0.3.12", default-features = false, features = ["std"] }
```

File: pbt-gen/Cargo.toml [hunk 2]
```diff
@@ -19,8 +19,8 @@ description = """
 default = ["time"]
 
 [dependencies]
-time = { version = "0.3", default-features = false, optional = true }
+time = { version = ">=0.3, <0.3.12", default-features = false, optional = true }
```
Comment (thanethomson) [inline pbt-gen/Cargo.toml:22]:
```suggestion
# TODO(thane): Remove restrictions once js-sys issue is resolved (causes the Substrate no_std check to fail)
time = { version = ">=0.3, <0.3.12", default-features = false, optional = true }
```

File: pbt-gen/Cargo.toml [hunk 3]
```diff
@@ -19,8 +19,8 @@ description = """
...
 proptest = { version = "0.10.1", default-features = false, features = ["std"] }
 
 [dev-dependencies]
-time = { version = "0.3", features = ["macros"] }
+time = { version = ">=0.3, <0.3.12", features = ["macros"] }
```
Comment (thanethomson) [inline pbt-gen/Cargo.toml:26]:
```suggestion
# TODO(thane): Remove restrictions once js-sys issue is resolved (causes the Substrate no_std check to fail)
time = { version = ">=0.3, <0.3.12", features = ["macros"] }
```

File: tendermint/Cargo.toml [hunk 4]
```diff
@@ -48,7 +48,7 @@ signature = { version = "1", default-features = false }
 subtle = { version = "2", default-features = false }
 subtle-encoding = { version = "0.5", default-features = false, features = ["bech32-preview"] }
 tendermint-proto = { version = "0.23.8", default-features = false, path = "../proto" }
-time = { version = "0.3", default-features = false, features = ["macros", "parsing"] }
+time = { version = ">=0.3, <0.3.12", default-features = false, features = ["macros", "parsing"] }
```
Comment (thanethomson) [inline tendermint/Cargo.toml:51]:
```suggestion
# TODO(thane): Remove restrictions once js-sys issue is resolved (causes the Substrate no_std check to fail)
time = { version = ">=0.3, <0.3.12", default-features = false, features = ["macros", "parsing"] }
```

File: testgen/Cargo.toml [hunk 5]
```diff
@@ -23,7 +23,7 @@ ed25519-dalek = { version = "1", default-features = false }
 gumdrop = { version = "0.8.0", default-features = false }
 simple-error = { version = "0.2.1", default-features = false }
 tempfile = { version = "3.1.0", default-features = false }
-time = { package = "time", version = "0.3", default-features = false, features = ["std"] }
+time = { package = "time", version = ">=0.3, <0.3.12", default-features = false, features = ["std"] }
```
Comment (thanethomson) [inline testgen/Cargo.toml:26]:
```suggestion
# TODO(thane): Remove restrictions once js-sys issue is resolved (causes the Substrate no_std check to fail)
time = { package = "time", version = ">=0.3, <0.3.12", default-features = false, features = ["std"] }
```

File: proto/Cargo.toml [hunk 6]
```diff
@@ -25,7 +25,7 @@ serde_bytes = { version = "0.11", default-features = false, features = ["alloc"]
 subtle-encoding = { version = "0.5", default-features = false, features = ["hex", "base64", "alloc"] }
 num-traits = { version = "0.2", default-features = false }
 num-derive = { version = "0.3", default-features = false }
-time = { version = "0.3", default-features = false, features = ["macros", "parsing"] }
+time = { version = ">=0.3, <0.3.12", default-features = false, features = ["macros", "parsing"] }
```
Comment (thanethomson) [inline proto/Cargo.toml:28]:
```suggestion
# TODO(thane): Remove restrictions once js-sys issue is resolved (causes the Substrate no_std check to fail)
time = { version = ">=0.3, <0.3.12", default-features = false, features = ["macros", "parsing"] }
```

**Review by thanethomson — APPROVED** [review]

**Pull request merged**